### PR TITLE
Update train_diffusion_dpo.py

### LINKED ISSUE #6702: [Diffusion DPO] Loss function and reproducing results in the paper
Hi, 

I got a question on the loss function used in both training scripts for diffusion DPO. 

In train_diffusion_dpo.py line 863 and train_diffusion_dpo_sdxl.py line 978
` loss = -1 * F.logsigmoid(inside_term.mean())`
From my understanding, this mean is the average taken across the batches as the empirical mean for formula (14) in the paper.
Screenshot 2024-01-24 at 3 54 39 PM
So I think the loss should instead be 
` loss = -1 * F.logsigmoid(inside_term).mean()`?

Also, may I know if there's anything else I need to modify other than changing the training configuration to reproduce the results in the paper, say 28.16 HPSv2 reward for the DPO-SDXL HPSv2 generations? Thank you very much!

### PR DESCRIPTION
**Pull request by viettmab**
# What does this PR do?

Fixes #6702 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue? #6702 
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@sayakpaul

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]
Thanks much.

Cc: @bram-w and @radames.

**Comment by sayakpaul** [comment]
Will merge after the CI is green.

**Comment by sayakpaul** [comment]
I think the failing tests are unrelated. Could you push an empty commit?

**Comment by viettmab** [comment]
@sayakpaul, I've pushed an empty commit as you suggested. Could you please check it again? Thanks!

**Comment by sayakpaul** [comment]
Failing test is unrelated.

**Comment by sayakpaul** [comment]
Thanks for your contributions.
